### ISSUE TITLE
Roller coaster building assist

### ISSUE THREAD

**Issue by whiterook6** [whiterook6/webgl#36]
As you build your coaster, the system should suggest the next sections of the track, either with a couple options or a single option. Perhaps the next few segments in a straight line or continuing the same path. Alternatively, the system could suggest a run of segments that keeps the G-forces consistent or minimized or below some limit.

To do / thoughts:
- come up with systems or strategies for how to suggest track
- does the whole track exist as a loop from the start, and building the coaster is really just deforming the loop?
- should this be more of a guided procedural generation, ala townscraper, or more manual, like RCT2?

Basically, how do we use a mouse and keyboard to build endlessly innovative track layouts?

**Comment by whiterook6** [comment]
Ideally people can be as obsessive over small details or as relaxed and casual as they like. I want to be able to precisely measure a corner's radius, bank angle, etc, but others may want to scribble an outline and have the computer do it.
